src/tfs/src/tfs.c

```diff
@@ -261,11 +261,20 @@ int tfsMkdirRecurAt(const char *rname, int level, int id) {
       // Try to create upper
       char *s = strdup(rname);
 
-      if (tfsMkdirRecurAt(dirname(s), level, id) < 0) {
-        tfree(s);
+      // Make a copy of dirname(s) because the implementation of 'dirname' differs on different platforms.
+      // Some platform may modify the contents of the string passed into dirname(). Others may return a pointer to
+      // internal static storage space that will be overwritten by next call. For case like that, we should not use
+      // the pointer directly in this recursion.
+      // See https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dirname.3.html
+      char *dir = strdup(dirname(s));
```

Comment:
1. The parameter 's' in dirname(s) is already a duplicated string of 'rname'. In this scenario, there is no other side effects even 's' is modified. 
2. Seems both free and tfree works. Why free() is a must?